napalm/nxos_ssh/nxos_ssh.py

```diff
@@ -1528,3 +1539,100 @@ def get_vlans(self):
+            # No metric present
+            vendor = vendor_re.search(port_ts).group(1)
+            vendor_part = vendor_part_re.search(port_ts).group(1)
+            vendor_rev = vendor_rev_re.search(port_ts).group(1)
+            serial_no = serial_no_re.search(port_ts).group(1)
+            type_s = type_no_re.search(port_ts).group(1)
+            # print(type_s)
+            state = {
+                "vendor": vendor.strip(),
+                "vendor_part": vendor_part.strip(),
+                "vendor_rev": vendor_rev.strip(),
+                "serial_no": serial_no.strip(),
+                "connector_type": self.connector_type_map.get(type_s, "Unknown"),
+            }
+            if "DOM is not supported" not in port_ts:
+                res = rx_instant_re.search(port_ts)
+                input_power = res.group(1) or res.group(2)
+                res = tx_instant_re.search(port_ts)
+                output_power = res.group(1) or res.group(2)
+                current = current_instant_re.search(port_ts).group(1)
+
+                # If interface is shutdown it returns "N/A" as output power
+                # or "N/A" as input power
+                # Converting that to -100.0 float
+                try:
+                    float(output_power)
+                except ValueError:
+                    output_power = -100.0
+                try:
+                    float(input_power)
+                except ValueError:
+                    input_power = -100.0
+
+                # Defaulting avg, min, max values to -100.0 since device does not
+                # return these values
+                optic_states = {
+                    "index": 0,
+                    "state": {
+                        "input_power": {
+                            "instant": (
+                                float(input_power) if "input_power" else -100.0
+                            ),
+                            "avg": -100.0,
+                            "min": -100.0,
+                            "max": -100.0,
+                        },
+                        "output_power": {
+                            "instant": (
+                                float(output_power) if "output_power" else -100.0
+                            ),
+                            "avg": -100.0,
+                            "min": -100.0,
+                            "max": -100.0,
+                        },
+                        "laser_bias_current": {
+                            "instant": (float(current) if "current" else -100.0),
+                            "avg": 0.0,
+                            "min": 0.0,
+                            "max": 0.0,
+                        },
+                    },
+                }
+                port_detail["physical_channels"]["channel"].append(optic_states)
+
+            port_detail["state"] = state
+            optics_detail[port] = port_detail
+            # print(port_detail)
```

Comment:
Can you remove this line please?

Reply:
Spurious print statements have been removed with the last commit.

Reply:
Updated with a sample test